File: sokoban/field.py
class Field:
    def __init__(self, key):
        self.visible = (key != None)
        if key == None:
            self.accessible = False
            self.target = False
            self.image = None
        elif key == '#':
            self.accessible = False
            self.target = False
            self.image = 'wall'
        elif key == ' ' or key == '$' or key == '@':
            self.accessible = True
            self.target = False
            self.image = 'floor'
        elif key == '.' or key == '*' or key == '+':
            self.accessible = True
            self.target = True
            self.image = 'target'

File: web-sokoban/field.js
export class Field {
    constructor(key) {
        this.visible = (key !== null)
        if (key === null) {
            this.accessible = false
            this.target = false
            this.image = null
        } else if (key === '#') {
            this.accessible = false
            this.target = false
            this.image = 'wall'
        } else if (key === ' ' || key === '$' || key === '@') {
            this.accessible = true
            this.target = false
            this.image = 'floor'
        } else if (key === '.' || key === '*' || key === '+') {
            this.accessible = true
            this.target = true
            this.image = 'target'
        }
    }
}
